lib/restforce/concerns/api.rb

```diff
@@ -344,7 +345,7 @@ def upsert!(sobject, field, attrs)
           fetch(attrs.keys.find { |k, v| k.to_s.downcase == field.to_s.downcase }, nil)
         attrs_without_field = attrs.
           reject { |k, v| k.to_s.downcase == field.to_s.downcase }
-        response = api_patch "sobjects/#{sobject}/#{field}/#{external_id}",
+        response = api_patch "sobjects/#{sobject}/#{field}/#{URI.encode(external_id)}",
```

Comment:
This looks sensible to me. I had a look through how encoding works, and elsewhere Faraday will do the encoding automatically.

@nagachika Can you add the same to the `#find` and `#select` methods (which also use an external ID)? Then I'll merge this in.